### PR TITLE
add model: regionvit

### PR DESCRIPTION
**Pull request by kaijieshi7**
## Add RegionViT
- [x] build model (F.unfold 算子不支持 https://github.com/Oneflow-Inc/oneflow/issues/3785)
- [x] update __init__.py in models
- [x] convert pretrained weight
- [x] inference test on imagenet and update model_zoo
- [x] update docs
- [x] update changelog 
- [x] pytorch speed comparison

### CODE REVIEW

**Comment by rentainhe** [comment]
可以先用nn.Unfold绕过去

**Comment by kaijieshi7** [comment]
测速时，196行加下面的代码
```python
for i, line in enumerate(lines):
    if "self.rel_pos_index = flow.tensor(out, dtype=flow.long)" in line:
        lines = lines[:i] + ["            self.rel_pos_index = flow.tensor(out)",] + lines[i + 1 :]
        break
for i, line in enumerate(lines):
    if "out.append(tmp[dy - y + offset_y, dx - x + offset_x].numpy())" in line:
        lines = lines[:i] + ["                            out.append(tmp[dy - y + offset_y, dx - x + offset_x])",] + lines[i + 1 :]
        break

```

**Comment by kaijieshi7** [comment]
OneFlow regionvit_tiny_224 time: 75.4ms (= 7537.5ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_tiny_224 time: 101.5ms (= 10150.3ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.35 (= 101.5ms / 75.4ms)

OneFlow regionvit_small_224 time: 87.5ms (= 8751.0ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_small_224 time: 126.7ms (= 12673.1ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.45 (= 126.7ms / 87.5ms)

OneFlow regionvit_small_w14_224 time: 168.0ms (= 16804.1ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_small_w14_224 time: 198.2ms (= 19815.2ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.18 (= 198.2ms / 168.0ms)

OneFlow regionvit_small_w14_peg_224 time: 178.5ms (= 17849.5ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_small_w14_peg_224 time: 224.2ms (= 22421.7ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.26 (= 224.2ms / 178.5ms)

OneFlow regionvit_medium_224 time: 139.1ms (= 13908.7ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_medium_224 time: 151.9ms (= 15191.3ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.09 (= 151.9ms / 139.1ms)

OneFlow regionvit_base_224 time: 166.2ms (= 16623.4ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_base_224 time: 201.1ms (= 20114.3ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.21 (= 201.1ms / 166.2ms)

OneFlow regionvit_base_w14_224 time: 299.6ms (= 29962.9ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_base_w14_224 time: 318.6ms (= 31858.9ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.06 (= 318.6ms / 299.6ms)

OneFlow regionvit_base_w14_peg_224 time: 305.8ms (= 30583.4ms / 100, input_shape=[16, 3, 224, 224])
PyTorch regionvit_base_w14_peg_224 time: 350.2ms (= 35016.8ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.14 (= 350.2ms / 305.8ms)

OneFlow regionvit_tiny_224 time: 75.9ms (= 7591.4ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_tiny_224 time: 189.1ms (= 18911.1ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 2.49 (= 189.1ms / 75.9ms)

OneFlow regionvit_small_224 time: 87.8ms (= 8775.0ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_small_224 time: 265.0ms (= 26503.2ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 3.02 (= 265.0ms / 87.8ms)

OneFlow regionvit_small_w14_224 time: 168.5ms (= 16848.2ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_small_w14_224 time: 433.3ms (= 43331.0ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 2.57 (= 433.3ms / 168.5ms)

OneFlow regionvit_small_w14_peg_224 time: 206.7ms (= 20672.3ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_small_w14_peg_224 time: 506.3ms (= 50626.8ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 2.45 (= 506.3ms / 206.7ms)

OneFlow regionvit_medium_224 time: 141.2ms (= 14120.5ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_medium_224 time: 309.5ms (= 30951.7ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 2.19 (= 309.5ms / 141.2ms)

OneFlow regionvit_base_224 time: 166.3ms (= 16634.6ms / 100, input_shape=[16, 3, 224, 224], world size=2)
PyTorch regionvit_base_224 time: 442.6ms (= 44260.8ms / 100, input_shape=[16, 3, 224, 224], world size=2)
✔️ Relative speed: 2.66 (= 442.6ms / 166.3ms)

File: flowvision/models/regionvit.py [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+import oneflow.nn.functional as F
+
+from flowvision.layers import LayerNorm2d, DropPath, Mlp, trunc_normal_
+from .helpers import to_2tuple
+from .utils import load_state_dict_from_url
+from .registry import ModelCreator
+
+
+model_urls = {
+    "tiny": "https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/flowvision/classification/RegionViT/regionvit_tiny_224.zip",
```
Comment (rentainhe) [inline flowvision/models/regionvit.py:19]: 这里是不是改成region_vit_tiny更合适一些呢, 我感觉这里的名字用全一点比较好

**Review by thinksoso — APPROVED** [review]

File: docs/source/changelog.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 - Add `resmlp_12_224_dino` model and pretrained weight [#128](https://github.com/Oneflow-Inc/vision/pull/128)
 - Support [ConvNeXt](https://arxiv.org/abs/2201.03545) model [#93](https://github.com/Oneflow-Inc/vision/pull/93)
 - Add [ReXNet](https://arxiv.org/abs/2007.00992) weights [#132](https://github.com/Oneflow-Inc/vision/pull/132)
+- Support RegionViT [#144](https://github.com/Oneflow-Inc/vision/pull/144)
```
Comment (rentainhe) [inline docs/source/changelog.md:13]: 这个ChangeLog可以先不加，之后我们会开一个V0.2.0的ChangeLog，加到新的里面

File: flowvision/models/regionvit.py [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+    if pretrained:
+        state_dict = load_state_dict_from_url(model_urls[arch], progress=progress)
+        model.load_state_dict(state_dict)
+    return model
+
+
+@ModelCreator.register_model
+def regionvit_tiny_224(pretrained=False, progress=True, **kwargs):
+    """
+    Constructs the Regionvit-tiny-224 model.
```
Comment (rentainhe) [inline flowvision/models/regionvit.py:576]: `Regionvit-tiny-224` -> `RegionViT-Tiny-224`

Comment (rentainhe) [reply flowvision/models/regionvit.py:576]: 下面同理

File: flowvision/models/regionvit.py (hunk 1 above)
Comment (kaijieshi7) [reply flowvision/models/regionvit.py]: 改了

File: docs/source/changelog.md (hunk 2 above)
Comment (kaijieshi7) [reply docs/source/changelog.md]: 取消了

File: flowvision/models/regionvit.py (hunk 3 above)
Comment (kaijieshi7) [reply flowvision/models/regionvit.py]: 改了

**Review by rentainhe — APPROVED** [review]

**Pull request merged**